src/test/daml/Daml/Finance/Asset/Test/IbericoPig.daml

```diff
@@ -0,0 +1,147 @@
+module Daml.Finance.Asset.Test.IbericoPig where 
+
+
+import ContingentClaims.Claim (Claim(..), one, scale)
+import ContingentClaims.Observation (Observation(..))
+import Daml.Finance.Interface.Asset.Instrument qualified as Instrument (I, View(..), disclosureUpdateReference)
+import Daml.Finance.Interface.Asset.Types (Id(..), InstrumentKey(..))
+import Daml.Finance.Interface.Common.Classes (toUTCTime)
+import Daml.Finance.Interface.Common.Disclosure qualified as Disclosure (I, SetObservers(..), View(..))
+import Daml.Finance.Interface.Common.Types (Observers)
+import Daml.Finance.Interface.Derivative.Types (Deliverable, TaggedClaim, C, taggedClaim)
+import Daml.Finance.Interface.Common.Util (flattenObservers)
+import Daml.Finance.Interface.Lifecycle.Lifecyclable qualified as Lifecyclable (I, Lifecycle(..), View(..))
+import Daml.Finance.RefData.Time.DateClock (Unit(..))
+import Daml.Finance.Interface.Derivative.Util.Claims.Lifecycle qualified as Lifecycle (lifecycle, splitPending, timeEvent)
+import Daml.Finance.Interface.Derivative.Util.Claims(toTime')
+import Daml.Finance.Interface.Derivative.HasClaims qualified as HasClaims (I, View(..))
+import Daml.Finance.Lifecycle.Effect (Effect(..))
+import Daml.Finance.Interface.Lifecycle.Event qualified as Event (I, View(..), getEventTime)
+import DA.Set (singleton)
+import DA.Text (sha256)
+import DA.Date (toDateUTC)
+
+data State = Pig 
+  | Butchered 
+  | Salted 
+  | Aged -- todo add age with int different int will have diff price 
+  | Restaurant 
+  | Eaten
+  deriving (Show, Eq)
+
+template StateTransitionEvent 
+  with 
+    currentOwner: Party
+    newOwner: Party
+    eventTime: Time
+    currentState: State
+  where
+    signatory currentOwner, newOwner
+
+    implements Event.I where
+      view = Event.View with eventTime
+
+
+template IbericoPig 
+  with 
+    farmer: Party
+    -- ^ The farmer that raised the pig
+    id: Id
+    -- ^ An identifier of the pig as an instrument 
+    -- (version will be the hash of the state that this pig is currently in
+    observers: Observers
+    -- ^ observers of the Iberico Pig, visible to public
+    state: State
+    -- ^ Current state of the pig, need to be in sequence
+
+    matureDate: Date
+    -- ^ Date that the farmer deemed the the pig has mature
+
+    cashInstrumentCid : Deliverable
+      -- ^ The cash instrument used to pay for one unit of pig.
+
+  where
+    signatory farmer -- for simplicity we will use single signatory in this example
+    observer flattenObservers observers
+
+    let 
+      instrumentKey = InstrumentKey with depository = farmer; issuer = farmer; id
+      
+      getClaimFromState: State -> Claim Date Decimal Deliverable Text -- todo I don't really need time/date
```

Comment:
For the flow, I don't really need TimedEvent, I am using StateTransitionEvent, which can happen any time. 
I feel like I am doing the wrong thing try to model this kind of non TimedEvent with `HasClaim` interface

Reply:
You are right that contingent claims is a powerful modelling framework to describe a payoff that depends on
* time
* elections from one of the parties

In the Iberico example most transitions from one state to the next do not depend on either time or election, which makes the framework less powerful as most contracts are of type `one pig` or `one pigLeg`.

Where I could see time kicking in is in the aging process, where I assume that the longer meat is aged, the more valuable it becomes. 

You could see this as a contract `anytime (t >= start) [ Give (one ham) `and` scale ( appreciationFunction (one USD) ) ` where `appreciationFunction` is an increasing function of time.